tests/platform_tests/daemon/test_thermalctld.py

```diff
@@ -0,0 +1,208 @@
+
+daemon_name = "thermalctld"
+
+SIG_STOP_SERVICE = None
+SIG_TERM = "-15"
+SIG_KILL = "-9"
+
+STATE_DB = 6
+thermalctld_tbl_key = ""
+
+@pytest.fixture(scope="module", autouse=True)
+def setup(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    daemon_en_status = check_pmon_daemon_enable_status(duthost, daemon_name)
+    if daemon_en_status is False:
+        pytest.skip("{} is not enabled in {}".format(daemon_name, duthost.facts['platform'], duthost.os_version))
+
+
+@pytest.fixture(scope="module", autouse=True)
+def teardown_module(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    yield
+
+    daemon_status, daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    if daemon_status is not "RUNNING":
+        duthost.start_pmon_daemon(daemon_name)
+        time.sleep(10)
+    logger.info("Tearing down: to make sure all the critical services, interfaces and transceivers are good")
+    check_critical_processes(duthost, watch_secs=10)
+
+
+@pytest.fixture
+def check_daemon_status(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    daemon_status, daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    if daemon_status is not "RUNNING":
+        duthost.start_pmon_daemon(daemon_name)
+        time.sleep(10)
+
+def check_expected_daemon_status(duthost, expected_daemon_status):
+    daemon_status, _ = duthost.get_pmon_daemon_status(daemon_name)
+    return daemon_status == expected_daemon_status
+
+def collect_data(duthost):
+    keys = duthost.shell('sonic-db-cli STATE_DB KEYS "PHYSICAL_ENTITY_INFO|*"')['stdout_lines']
+
+    dev_data = {}
+    for k in keys:
+        data = duthost.shell('sonic-db-cli STATE_DB HGETALL "{}"'.format(k))['stdout_lines']
+        data = compose_dict_from_cli(data)
+        dev_data[k] = data
+    
+    return {'keys': keys, 'data': dev_data}
+    
+def wait_data(duthost):
+    class shared_scope:
+        data_after_restart = {}
+    def _collect_data():
+        shared_scope.data_after_restart = collect_data(duthost)
+        return bool(shared_scope.data_after_restart['data'])
+    thermalctld_pooling_interval = 60
+    wait_until(thermalctld_pooling_interval, 6, 0, _collect_data)
+    return shared_scope.data_after_restart
+
+@pytest.fixture(scope='module')
+def data_before_restart(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    data = collect_data(duthost)
+    return data
+
+
+def test_pmon_thermalctld_running_status(duthosts, rand_one_dut_hostname, data_before_restart):
+    """
+    @summary: This test case is to check thermalctld status on dut
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    daemon_status, daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    logger.info("{} daemon is {} with pid {}".format(daemon_name, daemon_status, daemon_pid))
+    pytest_assert(daemon_status == expected_running_status,
```

Comment:
Do we need a wait_until here, as thermalctld might be stopped by other case, and it could take time to start up and write date to DB.

Reply:
This is the first daemon test and it is expecting to be run without any issue. If the previous test or other process stop this daemon or if the daemon is not running for any reason, this test needs to catch the reason.